src/sparseml/pytorch/sparsification/pruning/scorer.py

```diff
@@ -97,11 +98,19 @@ class PruningParamsGradScorer(PruningParamsScorer, ABC):
     Adds extra abstraction for handling gradient sharing between parameters
 
     :param params: list of model Parameters to track and score
+    :param grad_sampler: optional instance of GradSampler used to collect gradients
+        before pruning.
     """
 
-    def __init__(self, params: List[Parameter]):
+    def __init__(
+            self, 
+            params: List[Parameter],
+            grad_sampler: Optional[GradSampler] = None,
```

Comment:
are we going to add any shared logic for this?

Reply:
I removed the grad sampler from here since this pathway isn't used. It may be worthwhile to create a shared pathway down the line if more gradient-based pruners get added.